text/0195-kubernetes-resource-detection.md

```diff
@@ -0,0 +1,104 @@
+     fieldRef:
+       fieldPath: metadata.name
+- name: KUBERNETES_POD_UID
+   valueFrom:
+     fieldRef:
+       fieldPath: metadata.uid
+- name: KUBERNETES_NAMESPACE_NAME
+  valueFrom:
+    fieldRef:
+      fieldPath: metadata.namespace
+- name: KUBERNETES_NODE_NAME
+   valueFrom:
+     fieldRef:
+       fieldPath: spec.nodeName
+```
+
+### User Journey: Getting Started on Kubernetes
+
+As a user new to Kubernetes and/or OpenTelemetry, I can copy-and-paste the above into my pod specification to make resource discovery work for me. I don't need to set up any complex infrastructure, know about the downward API or OpenTelemetry resource detection, or the specifics of the lauguage SDK I'm using to make it work.
+
+### User Journey: Admission Webhooks
+
+There are a number of existing Kubernetes policy engines that can enforce a consistent set of environment variables on pods (in addition to doing much more), such as [kyverno](https://github.com/kyverno/kyverno), [podpreset](https://github.com/redhat-cop/podpreset-webhook), [open policy agent](https://www.openpolicyagent.org/), etc., but they all use the same underlying Kubernetes mechanism: [Admisison Webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#admission-webhooks).
+
+As a cluster administrator, I can use my preferred policy agent or my own mutating admission webhook to add the OpenTelemetry-specified set of environment variables to all pods subsequently created in the cluster. Application developers deploying to the cluster just need to use the OpenTelemetry SDK with the Kubernetes resource detector.
+
+### User Journey: Kubernetes Config Management
+
+Kubernetes templating tools, such as [Helm templates](https://helm.sh/docs/chart_template_guide/), allow users to easily re-use snippets of yaml in many places in their configuration.  A common set of environment variables would users to define the OpenTelemetry env var config once, and use it across all of their OpenTelemetry deployments.
+
+**Note:** This pattern unfortunately isn't easy to use with kubernetes' built-in patching tools, [kustomize](https://kustomize.io/), because it doesn't [currently](https://github.com/kubernetes-sigs/kustomize/issues/1493) support patching all containers across deployments with a single patch.
+
+## Internal details
+
+Add an experimental k8s.md specification under specification/resource, which specifies how kubernetes resource detectors should detect the kubernetes semantic conventions:
+
+SDK's should add a Kubernetes environment variable-based resource detector.  The Kubernetes resource detector and the collector's `resourcedetectionprocessor` should support the following environment variables, and map them to the following semantic conventions:
+
+| Environment Variable | Semantic Convention |
+| ---- | ---- |
+| K8s_POD_NAME | k8s.pod.name |
+| K8s_POD_UID | k8s.pod.uid |
+| K8s_NAMESPACE_NAME | k8s.namespace.name |
+| K8s_NODE_NAME | k8s.node.name |
```

Comment:
We should probably use al capital letters in env vars? also other mentions of these env vars need to be updated  in this doc.

```suggestion
| K8S_POD_NAME | k8s.pod.name |
| K8S_POD_UID | k8s.pod.uid |
| K8S_NAMESPACE_NAME | k8s.namespace.name |
| K8S_NODE_NAME | k8s.node.name |
```